Lib/platform.py

```diff
@@ -1167,6 +1167,14 @@ def platform(aliased=0, terse=0):
     if aliased:
         system, release, version = system_alias(system, release, version)
 
+    if system == 'Darwin':
+        # macOS (darwin kernel)
+        macos_release = mac_ver()[0]
+        if macos_release:
+            # note: 'macOS' is different than 'MacOS' used below
```

Comment:
This is something for another pull request, but the 'MacOS' bits you mention here are a leftover from the long removed MacOS 9 port and should be removed.